firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/handleFailedTransactions.ts

```diff
@@ -0,0 +1,41 @@
+import * as firebase from "firebase-admin";
+import { FirestoreBigQueryEventHistoryTrackerConfig } from ".";
+
+if (!firebase.apps.length) {
+  firebase.initializeApp();
+  firebase.firestore().settings({ ignoreUndefinedProperties: true });
+}
+
+export default async (
+  rows: any[],
+  config: FirestoreBigQueryEventHistoryTrackerConfig,
+  e: Error
+): Promise<void> => {
+  const db = firebase.firestore();
+  const batchArray = [db.batch()];
+
+  let operationCounter = 0;
+  let batchIndex = 0;
+
+  rows.forEach((row) => {
+    var ref = db.collection(config.backupTableId).doc(row.insertId);
+
+    batchArray[batchIndex].set(ref, {
+      ...row,
+      error_details: e.message,
+    });
+
+    operationCounter++;
+
+    // Check if max limit for batch has been met.
+    if (operationCounter === 499) {
+      batchArray.push(db.batch());
+      batchIndex++;
+      operationCounter = 0;
+    }
+  });
+
+  await batchArray.forEach(async (batch) => await batch.commit());
```

Comment:
This should be in a `for` loop since we're executing async logic:

```js
for (let batch in batchArray){
  await batch.commit();
}
```

Reply:
Batch can only be a typeOf `string` or `any` here?

`Property 'commit' does not exist on type 'string'.`

Would a for loop be a possible solution?